mlir/python/mlir/dialects/linalg/__init__.py

```diff
@@ -147,3 +147,51 @@ def __init__(
 
 
 generic = region_op(GenericOp_, terminator=YieldOp)
+
+
+def matmul(
+    inputs: Sequence[Union[Operation, OpView, Value]],
+    *,
+    outs: Sequence[Union[Operation, OpView, Value]],
+    indexing_maps: Sequence[AffineMapAttr],
```

Comment:
So my biggest confusion with the previous changes and the newly introduced requirements is: why do we suddenly need the user to provide indexing maps? I mean the indexing patterns for matmul are of course fixed and haven't changed so is it because we simply don't have implementations in C++ of those opdsl helpers that were prior computing these indexing maps in Python?

Reply:
Was just a copy-paste oversight - I started with ContractOp. Only when this was remedied did all the details of the arcane way to construct ops (from Python) start to matter.